update(GH): change image tag

zdtsw opened the pull request:
- commit on odh branch, should build image with tag "latest" -- same as now
- commit on main branch, should build image with tag "main" -- change from latest to main
- code coverage should run on "odh" branch -- no need run code coverage on main since it is a sync branch

this is a follow up https://github.com/opendatahub-io/llama-stack-k8s-operator/pull/18


<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Chores**
  * Updated workflows to adjust which branches trigger code coverage and image build processes.
  * Changed image tags in build workflows and environment configuration for improved consistency.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

VaishnaviHire commented:
Sorry I missed to address this comment on my PR.

/lgtm

leseb commented:
> Sorry I missed to address this comment on my PR.
> 
> /lgtm

That's ok! THanks!

zdtsw committed:
update(GH): change image names

- commit on odh should build image with tag "latest"
- commit on main should build image with tag "main"
- code coverage should run on "odh" branch

Signed-off-by: Wen Zhou <wenzhou@redhat.com>
